src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -175,6 +184,75 @@ func (c *controller) WaitAndUpdateNodesStatus(status *ControllerStatus) {
 	c.log.Infof("Done waiting for all the nodes. Nodes in error status: %d\n", hostsInError)
 }
 
+func (c *controller) HackDNSAddressConflict(wg *sync.WaitGroup) {
+
+	c.log.Infof("Making sure service %s can reserve the .10 address", dnsServiceName)
+
+	defer wg.Done()
+	networks, err := c.kc.GetServiceNetworks()
+	if err != nil || len(networks) == 0 {
+		c.log.Errorf("Failed to get service networks: %s", err)
+		return
+	}
+
+	ip, _, _ := net.ParseCIDR(networks[0])
+	ip4 := ip.To4()
+	if ip4 == nil {
+		c.log.Infof("Service network is IPv6: %s, skipping the .10 address hack", ip)
+		return
+	}
+	ip4[3] = 10 // .10 is the conflicting address
+
+	for i := 0; i < maxDNSServiceIPAttempts; i++ {
```

Comment:
can't we use it as function in retry?

Reply:
It will look convoluted, because we want to retry even when no errors occurred. I.e. find a conflicting service, delete it successfully, delete DNS pods successfully, and retry to make sure DNS service takes the IP. If it didn't - rinse, repeat. What error message will you write at the end of the loop in this case?